sos/report/plugins/coredump.py

```diff
@@ -8,36 +8,101 @@
 #
 # See the LICENSE file in the source distribution for further information.
 
+import os
+import re
 from sos.report.plugins import Plugin, IndependentPlugin, PluginOpt
 
 
 class Coredump(Plugin, IndependentPlugin):
+    """
+    This plugin collects from the systemd-coredump service, which is leveraged
+    via the coredumpctl command.
 
-    short_desc = 'Retrieve coredump information'
+    Beyond basic configuration of the service, the plugin will also attempt to
+    collect information on available coredumps. For any coredumps reported by
+    coredumpctl for which the core is present, this plugin will collect that
+    coredump file. The number of dump files collects is controlled by the
+    `dumps` option, defaulting to 5. Note that core collection is performed
+    based on the most recent cores generated. The summary information provided
+    by `coredumpctl info` is also collected.
+
+    The coredump files will be at their canonical location in the sos archive,
+    usually under var/lib/systemd/coredump. Symlinks are dropped in the plugin
+    directory to assist in identifying which coredump aligns to which entry in
+    the provided `coredumpctl list` output collection.
+
+    Users may leverage the `executable` option to control which coredumps the
+    plugin will collect. This option takes a case-insensitive python regex
+    string. If provided, only coredump entries for which the EXE field in
+    `coredumpctl list` output matches will be collected.
+
+    The dump files collected are compressed, and users should be aware that
+    when inflated these files can be orders of magnitude larger than their
+    collected sizes.
+    """
+
+    short_desc = 'systemd-coredump related information and dump files'
 
     plugin_name = "coredump"
     profiles = ('system', 'debug')
     packages = ('systemd-udev', 'systemd-coredump')
 
     option_list = [
-        PluginOpt("detailed", default=False,
-                  desc="collect detailed information for every report")
+        PluginOpt("dumps", default=5, desc="number of dump files to collect"),
```

Comment:
Collecting 5 coredumps as a default? Isn't it too much?

I am bit inclined to 3 as the default, but no strong preference (and no real use case / user experience how many coredumps are worth to take).

Reply:
5 was a shot in the dark number, no problem with dropping to 3.